### PR TITLE
Desktop Compliance Profiles Page Update

### PR DESCRIPTION
**Pull request by lancewf**
### :nut_and_bolt: Description: What code changed, and why?
Update the Compliance Profile page to include the case of the Desktop Offering is installed. This also updates wording and style for other cases. We want to hide and show the search bar, "X Profiles" "X Available" tabs, profile table, upload button, and text depending on if the Desktop Offering is installed, one or more profiles are installed, and the user has permissions to upload a profile. 

### :chains: Related Resources
#3749

### :+1: Definition of Done
All the below cases look correct. 

### :athletic_shoe: How to Build and Test the Change
1. `build components/automate-ui-devproxy && start_all_services && start_automate_ui_background && ui_logs`
1. Open https://a2-dev.test/compliance/compliance-profiles and ensure it looks like the below first case labeled "Non-Desktop, zero profiles, can upload profiles."
1. Select the "X Available" tab and image and profile.
1. Ensure the page looks like the below second case labeled "Non-Desktop, one or more profiles, can upload profiles."
1. Go to https://a2-dev.test/settings/users and create a user named "comp"
1. Go to https://a2-dev.test/settings/policies/compliance-viewer-access#members and add the user comp to the "Compliance Viewers" policy.
1. Log in as the comp user and go to the https://a2-dev.test/compliance/compliance-profiles page
1. Ensure the page looks like the below third case labeled "Non-Desktop, zero profiles, can not upload profiles."
1. Log back in as admin and remove the comp user from the "compliance-viewer-access" policy and add them to the "compliance-editor-access" policy. 
1. Log back in as the comp user and install a profile. 
1. Log back in as the admin user and move the comp user back to only the "compliance-viewer-access" policy. 
1. Log back in as the comp user and go to the profiles page https://a2-dev.test/compliance/compliance-profiles. 
1. Ensure the page looks like the below forth case labeled "Non-Desktop, one or more profiles, can not upload profiles."
1. Now we need to test these four cases with the Desktop offering installed. 
1. Run `enable_desktop` in hab studio and run through the above four cases. 

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
Cases:
- [x] Non-Desktop, zero profiles, can upload profiles.
 - Show search bar
 - Show tabs with the available profiles tab
 - Hide the profiles list
 - Show the upload profiles button.
 - Display the string "Get a profile from the Available tab or upload your own profile to get started!"
image

- [x] Non-Desktop, one or more profiles, can upload profiles.
 - Show search bar
 - Show tabs with the available profiles tab
 - Show profiles list
 - Show the upload profiles button. 
image

- [x] Non-Desktop, zero profiles, can not upload profiles.
 - Show search bar
 - Show tabs with the available profiles tab
 - Hide the profiles list
 - Hide the upload profiles button.
 - Display the string "It looks like no one has uploaded any profiles yet and you don't have permission to upload them. If this is a mistake, then reach out to your administrator."
image

- [x] Non-Desktop, one or more profiles, can not upload profiles.
 - Show search bar
 - Show tabs with the available profiles tab
 - Show profiles list
 - Hide the upload profiles button. 
image

- [x] Desktop, zero profiles, can upload profiles.
 - Hide tabs with the available profiles tab. 
 - Hide profiles list
 - Show the upload profiles button. 
 - Display the string "Upload a profile to get started!"
image

- [x] Desktop, zero profiles, can not upload profile.
 - Hide tabs with the available profiles tab. 
 - Hide profiles list
 - Hide the upload profiles button. 
 - Display the string "It looks like no one has uploaded any profiles yet and you don't have permission to upload them.If this is a mistake, then reach out to your administrator."
image

- [x] Desktop, one or more profiles, can upload profiles.  
 - Hide tabs with the available profiles tab. 
 - Show profiles list
 - Show the upload profiles button. 
 - Show search bar
image

- [x] Desktop, one or more profiles, can not upload profile. 
 - Hide tabs with the available profiles tab. 
 - Show profiles list
 - Hide the upload profiles button. 
 - Show search bar
image

### CODE REVIEW

File: components/automate-ui/src/app/pages/+compliance/+profile/+profile-overview/profile-overview.component.ts [hunk 1]
```diff
@@ -138,6 +145,11 @@ export class ProfileOverviewComponent implements OnInit, OnDestroy {
 
   // load profiles available to the user
   loadAvailableProfiles() {
+    if (!this.isAvailableProfilesVisible) {
+      this.availableListLoading = false;
+      this.availableProfilesDataLoaded = true;
+      return of([]);
```
Comment (lancewf) [inline components/automate-ui/src/app/pages/+compliance/+profile/+profile-overview/profile-overview.component.ts:151]: Not loading the available profiles if they are not going to be displayed.

File: components/automate-ui/src/app/pages/+compliance/+profile/+profile-overview/profile-overview.component.ts [hunk 2]
```diff
@@ -72,15 +74,20 @@ export class ProfileOverviewComponent implements OnInit, OnDestroy {
...
     private availableProfilesService: AvailableProfilesService,
     private uploadService: UploadService,
     private chefSessionService: ChefSessionService,
-    private layoutFacade: LayoutFacadeService
-  ) {}
+    private layoutFacade: LayoutFacadeService,
+    private productDeployedService: ProductDeployedService
+  ) {
+    this.isAvailableProfilesVisible = !this.productDeployedService.isProductDeployed('desktop');
+  }
```
Comment (lancewf) [inline components/automate-ui/src/app/pages/+compliance/+profile/+profile-overview/profile-overview.component.ts:90]: If the Desktop offering is installed the available profiles should not be visible.

File: components/automate-ui/src/app/pages/+compliance/+profile/+profile-overview/profile-overview.component.html [hunk 3]
```diff
@@ -17,15 +17,15 @@
         <chef-heading>Profiles</chef-heading>
         <chef-subheading>Compliance profiles manage security and compliance scans.</chef-subheading>
 
-        <div class="profiles-search">
+        <div class="profiles-search" *ngIf="!profilesEmpty || isAvailableProfilesVisible">
```
Comment (lancewf) [inline components/automate-ui/src/app/pages/+compliance/+profile/+profile-overview/profile-overview.component.html:20]: Do not show the search bar if there are no install profiles and the available profiles are not visible.

File: components/automate-ui/src/app/pages/+compliance/+profile/+profile-overview/profile-overview.component.html [hunk 4]
```diff
@@ -17,15 +17,15 @@
...
           <input
             chefInput
             type="text"
             placeholder="Search profiles..."
             (input)="onSearchInput($event)"/>
         </div>
 
-        <chef-tab-selector class="profiles-tabs" [value]="selectedTab" (change)="onTabChange($event)">
+        <chef-tab-selector *ngIf="isAvailableProfilesVisible" class="profiles-tabs" [value]="selectedTab" (change)="onTabChange($event)">
```
Comment (lancewf) [inline components/automate-ui/src/app/pages/+compliance/+profile/+profile-overview/profile-overview.component.html:28]: Only show the two tabs image if the available profiles are visible.

File: components/automate-ui/src/app/pages/+compliance/+profile/+profile-overview/profile-overview.component.spec.ts [hunk 5]
```diff
@@ -43,13 +54,18 @@ describe('ProfilesOverviewComponent', () => {
         StoreModule.forRoot(ngrxReducers, { runtimeChecks })
       ],
       declarations: [
-        ProfileOverviewComponent
+        ProfileOverviewComponent,
+        // This returns true for all app-authorized checks on the page.
+        MockComponent({ selector: 'app-authorized',
+          inputs: ['allOf'],
+          template: '<ng-content *ngIf="true"></ng-content>' })
```
Comment (msorens) [inline components/automate-ui/src/app/pages/+compliance/+profile/+profile-overview/profile-overview.component.spec.ts:61]: Ooh! I should have thought of that--this is a great test fixture!!

File: components/automate-ui/src/app/pages/+compliance/+profile/+profile-overview/profile-overview.component.spec.ts [hunk 6]
```diff
@@ -35,296 +48,617 @@ describe('ProfilesOverviewComponent', () => {
...
-      ],
-      providers: [
-        {provide: ProfilesService, useClass: MockProfilesService},
-        {provide: ChefSessionService, useValue: mockSession},
-        {provide: UploadService, useClass: MockUploadService},
-        FeatureFlagsService
-      ],
-      schemas: [ CUSTOM_ELEMENTS_SCHEMA ]
+  describe('non-Desktop with user upload permissions', () => {
+    beforeEach(() => {
```
Comment (vjeffrey) [inline components/automate-ui/src/app/pages/+compliance/+profile/+profile-overview/profile-overview.component.spec.ts:52]: holy 746 test lines added!! 🎉

**Pull request merged**